.github/workflows/tests.yml

```diff
@@ -0,0 +1,54 @@
+name: tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 12 * * 0'  # run once a week on Sunday
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        config:
+        # [Python version, tox env]
+        - ["2.7",   "py27"]
+        - ["2.7",   "py27-minimal"]
+        - ["3.5",   "py35"]
+        - ["3.6",   "py36"]
+        - ["3.7",   "py37"]
+        - ["3.8",   "py38"]
+        - ["3.9",   "py39"]
+        - ["pypy2", "pypy"]
+        - ["pypy3", "pypy3"]
+        - ["3.8",   "coverage"]
```

Comment:
I deeply dislike running coverage on only a single version of Python.

It has almost universally resulted in coverage dropping.

It pretty much negates the benefits of coveralls aggregating coverage across the runs; the same effect has been achieved locally by  having an environment that depends on the others and does 'coverage combine' after running all the environments.

The coverage-python-version plugin is…fine...for 2/3 things, but less fine for 3.x vs 3.y. But we don't need it if we just always run coverage.

But...apparently that's what we're doing now, so this isn't this PRs fault. I'm just venting.

(I also deeply dislike using tox on CI, but that's a completely separate issue and possibly a personal hangup 😄 )

Reply:
About using `tox` on CI, the advantage I see is that the local test runs share the exact same configuration with the external CI service. Only one place to look, one place to make changes and less of a chance that the configurations drift apart.

Reply:
There is some DRY benefit to just calling out to tox.ini, I agree. But it doesn't seem significant to me. In almost all cases where you modify tox.ini, you're still going to need to modify the CI file to add the new environment (simply changing the one command that invokes the testrunner, one time you would *only* need to modify tox.ini, is quite rare). True, GHA currently makes it annoyingly hard (verbose) to run slightly modified test environments, so there is some additional benefit there that Travis and Appveyor don't need.

The downsides I see are extra time and complexity on CI: We're creating a virtual environment inside the virtual environment that CI has already created for us, and we're using another tool that introduces its own potential set of bugs or incompatibilities.

But it's six of one, half a dozen of the other (as the saying goes). I'm not fighting it, I just don't like it. 🤷